core/rawdb/accessors_state_sync.go

```diff
@@ -191,3 +191,32 @@ func GetLatestSyncPerformed(db ethdb.Iteratee) uint64 {
 	}
 	return latestSyncPerformed
 }
+
+// clearPrefix removes all keys in db that begin with prefix and match an
+// expected key length. `keyLen` should include the length of the prefix.
+func clearPrefix(db ethdb.KeyValueStore, prefix []byte, keyLen int) error {
+	it := db.NewIterator(prefix, nil)
+	defer it.Release()
+
+	batch := db.NewBatch()
+	for it.Next() {
+		key := common.CopyBytes(it.Key())
+		if len(key) != keyLen {
+			// avoid deleting keys that do not match the expected length
```

Comment:
```suggestion
```

It's already stated in the function comment and is obvious from context.

Reply:
Thanks for the suggestion, bundled in 1ccca3b0bdaf5fc3516a95d60596678a3ee78738.
Note these are simply code movements of existing code, but why not change them whilst we're at it indeed 👍